### PR TITLE
Supports IEEE Floating Point Number Datatypes

### PR DESCRIPTION
**Pull request by kaimal11**
Implements xsd::Float and xsd::Double datatypes

### CODE REVIEW

File: src/rdf4cpp/rdf/datatypes/xsd/Double.hpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ */
+template<>
+inline std::string LiteralDatatypeImpl<xsd_double>::to_string(const cpp_type &value) {
+
+    if (isnan(value)) {
+        //checks whether value is "nan", then returns rdf representation of nan
+        return "NaN";
+    } else if (isinf(value)) {
+        //checks whether value is "inf", then returns rdf representation of inf
+        return "INF";
```
Comment (nkaralis) [inline src/rdf4cpp/rdf/datatypes/xsd/Double.hpp]: We need to also handle -INF.

File: src/rdf4cpp/rdf/datatypes/xsd/Float.hpp [hunk 2]
```diff
@@ -41,17 +47,40 @@ inline LiteralDatatypeImpl<xsd_float>::cpp_type LiteralDatatypeImpl<xsd_float>::
...
-    str_os << value;
-    // Get string from output string stream
-    std::string str = str_os.str();
-    return str;
+    if (isnan(value)) {
+        //checks whether value is "nan", then returns rdf representation of nan
+        return "NaN";
+    } else if (isinf(value)) {
+        //checks whether value is "inf", then returns rdf representation of inf
+        return "INF";
```
Comment (nkaralis) [inline src/rdf4cpp/rdf/datatypes/xsd/Float.hpp:58]: As in Double.hpp, we need to also handle -INF.

File: tests/datatype/tests_Double.cpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    auto lit8 = Literal{"NaN", type_iri};
+    CHECK(isnan(lit8.value<datatypes::xsd::Double>()));
+
+    auto lit9 = Literal{"INF", type_iri};
+    CHECK(isinf(lit9.value<datatypes::xsd::Double>()));
+
+    value = INFINITY;
+    auto lit10 = Literal::make<datatypes::xsd::Double>(value);
+    CHECK(isinf(lit10.value<datatypes::xsd::Double>()));
```
Comment (nkaralis) [inline tests/datatype/tests_Double.cpp:62]: We need to also check for -INF

**Review by nkaralis — CHANGES_REQUESTED** [review]

**Comment by bigerl** [comment]
we should have another look at precision before we merge 💡

**Pull request closed without merging**